src/wallet/spend.cpp

```diff
@@ -987,14 +991,21 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
 
     const CAmount change_amount = result.GetChange(coin_selection_params.min_viable_change, coin_selection_params.m_change_fee);
     if (change_amount > 0) {
-        CTxOut newTxOut(change_amount, scriptChange);
-        if (nChangePosInOut == -1) {
-            // Insert change txn at random position:
-            nChangePosInOut = rng_fast.randrange(txNew.vout.size() + 1);
-        } else if ((unsigned int)nChangePosInOut > txNew.vout.size()) {
-            return util::Error{_("Transaction change output index out of range")};
+        if (existent_change_out_index) {
+            // If there is an output with the change address already, increase that one instead of creating a new output
+            txNew.vout.at(*existent_change_out_index).nValue += change_amount;
```

Comment:
I would expect `GetChange(…)` to deduct the fee for the change, but if you already have a change output, wouldn’t that mean that you account for the change fee twice? You might need to add the change_fee back in the case where you just increase an existing change output.

Reply:
It depends on how the output was crafted. If it was created manually, it will not account for the fee. And also, it will not account for it if it was created in a previous `createTransaction` call with SFFO enabled.

It is fine to "overpay" the fee at this line because later in the process (50 lines below) we check if the final fee is above the needed one, automatically increasing the change by the surplus if needed.

Reply:
Isn't the fee for all outputs accounted for in `not_input_fees`? It seems to me this will always overpay and then will be corrected when we detect surplus fee.

Maybe a clearer way to handle this would be to pass `chnage_fee=0` to `GetChange` if `existent_change_out_index=true`. I'd prefer to minimise number of cases when we need to correct fees at the later stage of transaction building.